agent/xds/endpoints.go

```diff
@@ -155,6 +157,51 @@ func (s *Server) endpointsFromSnapshotMeshGateway(cfgSnap *proxycfg.ConfigSnapsh
 		resources = append(resources, la)
 	}
 
+	// generate the endpoints for the service subsets
+	for svc, resolver := range cfgSnap.MeshGateway.ServiceResolvers {
+		for subsetName, subset := range resolver.Subsets {
+			clusterName := ServiceSNI(svc, subsetName, "default", cfgSnap.Datacenter, cfgSnap)
+
+			endpoints := cfgSnap.MeshGateway.ServiceGroups[svc]
+
+			// locally execute the subsets filter
+			filterExp := subset.Filter
+			if subset.OnlyPassing {
+				// we could do another filter pass without bexpr but this simplifies things a bit
+				if filterExp != "" {
+					// TODO (filtering) - Update to "and all Checks as chk { chk.Status == passing }"
+					//                    once the syntax is supported
+					filterExp = fmt.Sprintf("(%s) and not Checks.Status != passing", filterExp)
+				} else {
+					filterExp = "not Checks.Status != passing"
```

Comment:
Can we express that without a double negative? Is `Check.Status == passing` equivalent?

Reply:
Unfortunately no, I am writing up a proposal to change the bexpr language a bit to allow this to be a little more natural.

The problem is that `Checks.Status == passing` will eval to true if *any* check is passing. The double negative is needed to ensure that *all* checks are passing.

I am thinking the way to go is to pull in the Sentinel languages any/all expressions for this to become:

`all Checks as chk { chk.Status == passing }`

While it doesn't help much with the length/verbosity of the expression its much clearer what the intent is. The any/all expressions also have other important uses or else we could just implement `all Checks.Status == passing`. The main thing that doesn't solve is the ability to perform operations on collections of primitives. So if you have a `[]string`  or a `map[string]string` and just want to know if all the values within match a regex for example you cannot do it (even once we have regex/substring capabilities which I have PRs open for).

Anyways, I will be writing some stuff up about this in the coming weeks as @johncowen is going to need the functionality for the UI.